### ISSUE TITLE
[IndentationCheck] 'new' has incorrect indentation level (inline variable assignment)

### ISSUE THREAD

**Issue by stechio** [checkstyle/checkstyle#13405]
I have read check documentation: https://checkstyle.org/checks/misc/indentation.html#Indentation
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
$ javac Test.java

$ cat test-checkstyle.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name = "Checker">
  <module name="TreeWalker">
    <module name="Indentation">
      <property name="basicOffset" value="2"/>
      <property name="lineWrappingIndentation" value="4"/>
    </module>
  </module>
</module>

$ cat Test.java 
public class Test {
  public static void main(String[] args) {
    java.util.List<String> list = new java.util.ArrayList<>();

    list.add(
        new String("A long long long long string to force line-wrapping on my code formatter"));

    String v;
    list.add(v =
        new String("A long long long long string to force line-wrapping on my code formatter"));
  }
}

$ java -jar checkstyle-10.12.1-all.jar -c test-checkstyle.xml Test.java
Starting audit...
[ERROR] Test.java:10:9: 'new' has incorrect indentation level 8, expected level should be one of the following: 10, 12. [Indentation]
Audit done.
Checkstyle ends with 1 errors.
```
---
CURRENT BEHAVIOR
In case of inline variable assignment, the indentation check doesn't honor the expected line-wrapping rules.

---
EXPECTED BEHAVIOR
Line wrapping indentation on inline variable assignment should be treated the same as any other.

**Comment by romani** [comment]
@stechio , such assignment is redundant.
```
list.add(v = new String("A long..."));
```

Conceptually you are right that Check should not care about this, 
.... but please share what is reasonable real life use case for such redundant assignment ?

there infinite amount of ways to write some weird code, we can not cover all possible, we usually forcus more on reasonable first and if have time to unreasonable but possible.

**Comment by stechio** [comment]
Please, keep in mind that my `Test.java` is just a distilled representation of a real-life use case, stripped off of anything unnecessary to reproduce the issue (`Test.java` is not supposed to do anything meaningful, but to trigger checkstyle to spit its false error).

My real-life use case is about the instantiation of a domain-specific object (which here I had to replace with `String` for obvious reasons) and its assignment to a local variable; such assignment is inlined into a `java.util.Map.put(..)` call for convenience (quite the contrary of "redundancy", and definitely not "weird" code: because of `ConcurrentModificationException`, I cannot use `java.util.Map.computeIfAbsent(..)`, so in case of missing entry I have to `put` and assign at the same time).

Furthermore, whilst checkstyle returns ERROR on this...

```java
list.add(v =
  new String("A long..."));
```

...it returns SUCCESS on this:

```java
list.add(v =
  "A long...");
```

So, *it's not a problem of inline assignment per-se, just of inlined `new` operator*... THAT lack of support is weird, NOT the other way around!

**Comment by romani** [comment]
I know that we by default asking for dumb code to see a problem.

But in this case I really curious when and it it good to write like this. So it will be shared with others.
Contributors usually select issue to resolve they understand and know why they are doing it. If issue is very artificial, contributors have less intensives to fix it.

If code is very proprietary, you can obfuscate it a bit, but still show a reason.
